src/serialize.c

```diff
@@ -123,6 +120,17 @@ btc_bool deser_u16(uint16_t* vo, struct const_buffer* buf)
     return true;
 }
 
+btc_bool deser_i32(int32_t* vo, struct const_buffer* buf)
+{
+    int32_t v;
```

Comment:
instead of copying the function implementation, we should use `deser_u16((uint32_t) v_, buf);`